lib/components/primitive-components/Group/Group.ts

```diff
@@ -56,6 +68,91 @@ export class Group<Props extends z.ZodType<any, any, any> = typeof groupProps>
     return {
       zodProps: groupProps as unknown as Props,
       componentName: "Group",
+      shouldRenderAsSchematicBox: Boolean(
+        (this._parsedProps as any)?.showAsSchematicBox ??
+          (this.props as any)?.showAsSchematicBox ??
+          false,
+      ),
+    }
+  }
+
+  private _isShowAsSchematicBoxEnabled(): boolean {
+    return Boolean(this._parsedProps.showAsSchematicBox)
+  }
+
+  private _resolvePortFromSelector(selector: string): Port | null {
+    const normalized = selector.trim()
+    if (!normalized) return null
+
+    const direct = this.selectOne(normalized, { type: "port" }) as Port | null
+    if (direct) return direct
+
+    const dotIndex = normalized.lastIndexOf(".")
+    if (dotIndex === -1) return null
+
+    const parentSelector = normalized.slice(0, dotIndex)
+    const portToken = normalized.slice(dotIndex + 1)
+
+    let parentComponent = this.selectOne(parentSelector)
+    if (!parentComponent && parentSelector && !/[.#\[]/.test(parentSelector)) {
+      parentComponent = this.selectOne(`.${parentSelector}`)
+    }
+    if (!parentComponent) return null
+
+    const ports = parentComponent.children.filter(
+      (c) => c.componentName === "Port",
+    ) as Port[]
+
+    return (
+      ports.find((port) => port.getNameAndAliases().includes(portToken)) ?? null
+    )
```

Comment:
it feels like this code is copied from somewhere and it's dangerous to repeat it since if there's a bug it won't be fixed in both places

You can either call the method of the implementing class i.e. `Port._resolvePortFromSelector(this, ...)` or you can extend a class (usually better)